app/src/ereferrals/java/org/eyeseetea/malariacare/data/database/datasources/SettingsDataSource.java

```diff
@@ -28,4 +38,36 @@ private Locale getCurrentLocale() {
             return Resources.getSystem().getConfiguration().locale;
         }
     }
+
+    private void setMediaPreference(String listType) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(
+                context);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString(context.getResources().getString(R.string.media_list_style_preference), listType);
+        editor.commit();
+    }
+
+    private String getMediaPreference() {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(
+                context);
+        return sharedPreferences.getString(
+                context.getResources().getString(R.string.media_list_style_preference),
+                "");
+    }
+
+    @Override
+    public MediaListMode getMediaListMode() {
```

Comment:
This method should not be public because this data source should return only settings domain entity